cpp/dolfinx/la/PETScMatrix.cpp

```diff
@@ -201,12 +201,7 @@ PETScMatrix::set_fn(Mat A, InsertMode mode)
 #else
     ierr = MatSetValuesLocal(A, m, rows, n, cols, vals, mode);
 #endif
-
-#ifdef DEBUG
```

Comment:
I found these useful in cases where `MatSetValuesLocal` went silently wrong in Release mode.

Reply:
I think they should be caught at the higher level by the caller.

The original PR turned on the check in all modes, which isn't appropriate.